src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -2505,4 +2543,8 @@ public String getFullName() {
   public void copyResource(String src, String dest) throws IOException {
     FileIO.copy(getResourceDir() + File.separator + src, dest);
   }
+
+  public void setId(String id) {

```

Comment:
just curious, why should you be able to set the id of a service?  seems like this is something that should be immutable?

Reply:
yeah ... very much agree ...
It was residue from trying to get serialization all worky 
Serialization needs to set the id (from remote systems) - now it does without this